### PR TITLE
Support language switching in Fides JS banner

### PR DESCRIPTION
**Pull request by gilluminate**
Closes: https://ethyca.atlassian.net/browse/PROD-1598

### Description Of Changes

When a banner or modal supports multiple translations of content, then an end user can switch the banner and modal between the available languages via a language picker.

### Code Changes

* [x] Adds ability to get available locales from the config
* [x] Adds new language switcher and styles to the button groups on the banner
* [x] Adds new MenuItem component
* [x] Adds new button on the demo page for deleting the cookie and reloading for convenience
* [x] Adds new language svg icon (from fontawesome)
* [x] Cleans up a few style issues on the existing banner
* [x] Adds new, simple i18n context/provider for updating state when user changes locale.

### Steps to Confirm

* [ ] visit http://localhost:3000/fides-js-components-demo.html
* [ ] hover over the language icon and select a new language from the list
* [ ] default (english) text should now be replaced by newly selected language text
* [ ] a11y: using the tab key, user should be able to access each of the language options without using a mouse

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [x] Update `CHANGELOG.md`
* [x] Update `README.md`

### CODE REVIEW

File: clients/privacy-center/cypress/e2e/consent-banner.cy.ts [hunk 1]
```diff
@@ -100,24 +100,27 @@ describe("Consent overlay", () => {
           ).contains("[banner-opts] We use cookies and similar methods");
           cy.get("div#fides-button-group").within(() => {
             cy.get(
-              "button#fides-banner-button-tertiary.fides-banner-button.fides-banner-button-tertiary"
```
Comment (NevilleS) [inline clients/privacy-center/cypress/e2e/consent-banner.cy.ts:103]: 👌

File: clients/privacy-center/cypress/e2e/consent-i18n.cy.ts [hunk 2]
```diff
@@ -694,16 +694,18 @@ describe("Consent i18n", () => {
...
       it(`localizes in the user selected locale (${SPANISH_LOCALE})`, () => {
         // Visit the demo site in English, but expect Spanish translations when the user selects
         visitDemoWithI18n({
           navigatorLanguage: ENGLISH_LOCALE,
           globalPrivacyControl: true,
           fixture: "experience_banner_modal.json",
         });
-        // TODO (PROD-1598): select Spanish from banner
+        // eslint-disable-next-line cypress/no-unnecessary-waiting
+        cy.wait(200); // wait for the overlay to appear
```
Comment (NevilleS) [inline clients/privacy-center/cypress/e2e/consent-i18n.cy.ts:706]: nit: I agree with eslint! You should trust Cypress to run-and-retry as much as possible!

**Review by NevilleS — APPROVED** [review]
Couple nitpicks here and there, but overall this looks great to me! 😄

File: clients/fides-js/src/components/Button.tsx [hunk 3]
```diff
@@ -4,19 +4,21 @@ import { ButtonType } from "../lib/consent-types";
...
   buttonType,
   label,
+  id,
   onClick,
   className = "",
 }) => (
   <button
     type="button"
-    id={`fides-banner-button-${buttonType.valueOf()}`}
+    id={id}
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/fides-js/src/components/Button.tsx:21]: Cool. Yeah, I was thinking it'd be nice to implement an e2e specific property, like `data-test` to help avoid them even further.

File: clients/privacy-center/cypress/e2e/consent-i18n.cy.ts (hunk 2 above)
Comment (gilluminate) [reply clients/privacy-center/cypress/e2e/consent-i18n.cy.ts:706]: generally I do too, but a) saw this used in other places where our cypress tests need to wait for the banner so assumed it was an acceptable exception. b) Cypress was running but not re-trying this, just times out and fails. (even waiting 100ms isn't long enough). Open to other solutions.

File: clients/fides-js/src/lib/i18n/i18n-context.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+interface I18nContextProps {
+  currentLocale: string;
+  setCurrentLocale: StateUpdater<string>;
+}
+
+const I18nContext = createContext<I18nContextProps>({} as I18nContextProps);
+
+export const I18nProvider: FunctionComponent = ({ children }) => {
+  const { locale } = i18n;
+  const [currentLocale, setCurrentLocale] = useState(locale);
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/fides-js/src/lib/i18n/i18n-context.tsx:14]: It's a bit of a hack, if I'm honest. It does work, but [you'll notice](https://github.com/ethyca/fides/pull/4737/files#diff-5f38f505be8a501cc34baebb4cca30819539c21aa0ab8a2cb97fe71581a3617eR38-R39) I do _both_ when the language selection is made. The only (current) place the hook is used to consume the property in the `useEffect`s of the NoticeOverlay to update those `bestTranslation` instances. I think this mimics LinguiJS as closely as possible, without access to their `.on()` event emitter.

File: clients/fides-js/src/components/Button.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/fides-js/src/components/Button.tsx:21]: Oh! There already is one :)
`data-testid`

File: clients/fides-js/src/lib/i18n/i18n-context.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  const value: I18nContextProps = useMemo(
+    () => ({ currentLocale, setCurrentLocale }),
+    [currentLocale]
+  );
+  return <I18nContext.Provider value={value}>{children}</I18nContext.Provider>;
+};
+export const useI18n = () => {
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/fides-js/src/lib/i18n/i18n-context.tsx:22]: very open to this level of scrutiny! love it.

File: clients/fides-js/src/lib/i18n/i18n-context.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (NevilleS) [reply clients/fides-js/src/lib/i18n/i18n-context.tsx:14]: OK, sounds good!

**Pull request merged**